.github/workflows/LocalTesting.yml

```diff
@@ -95,4 +99,10 @@ jobs:
           ICEBERG_SERVER_AVAILABLE: 1
           DUCKDB_ICEBERG_HAVE_GENERATED_DATA: 1
         run: |
-          make test_release
\ No newline at end of file
+          make test_release
+
+      - name: Test reads with PyIceberg
+        env:
+          ICEBERG_SERVER_AVAILABLE: 1
+        run: |
+          python3 -m pytest test/python
```

Comment:
great idea to have CI test reading the written data with other engines

I think it would be nice to make this flow a bit more explicit where we write separate tests to generate the data for testing in pyiceberg and spark. Currently it would we really easy to break ci by refactoring a test in `test/sql` which you would only catch when running CI.

For now I would say this is fine, but as more of these tests would get added i think its nice to at least group all tests that generate data for reading through pyiceberg or spark in a single folder with a readme explaining how they're tested

Reply:
Hmm I agree, but I don't see a nice way of grouping these

What we *could* do is not have a `.test` for the insert related tests but implement these in `test/python` using duckdb-python, then they're bundled and we can potentially even do a clean up after the tests are ran to make it rerunnable (which currently requires manual intervention to clean up the files:
```bash
python3 -m scripts.data_generators.generate_data spark-rest --test insert_test
python3 -m scripts.data_generators.generate_data spark-rest --test insert_all_types
```
)

Reply:
Let's take a look at making this cleaner in a future PR